src/ui/public/modals/modal_overlay.js

```diff
@@ -5,16 +5,23 @@ import modalOverlayTemplate from './modal_overlay.html';
  * Appends the modal to the dom on instantiation, and removes it when destroy is called.
  */
 export class ModalOverlay {
-  constructor(modalElement) {
+  constructor(modalElement, scope) {
     this.overlayElement = angular.element(modalOverlayTemplate);
     this.overlayElement.append(modalElement);
+
+    angular.element(document.body).on('keydown', (event) => {
+      if(event.keyCode === 27) {
+        scope.onCancel();
```

Comment:
Fair enough.  Actually the whole reason for the onClose functionality may be going away, so we could rip that out (though I'd prefer to wait till after https://github.com/elastic/kibana/pull/10585 gets checked in just to be sure).

My modified feedback, in that case, is that I still think this logic should either go in confirm_modal, or be passed in as a function.  No need for the modal overlay to have scope knowledge, and this could easily break if we change the name of the function inside confirm_modal.